docs/requirements-insiders.txt

```diff
@@ -4,7 +4,7 @@ mkdocs==1.6.1
 mkdocs-material @ git+ssh://git@github.com/astral-sh/mkdocs-material-insiders.git@39da7a5e761410349e9a1b8abf593b0cdd5453ff
 mkdocs-redirects==1.2.2
 mdformat==0.7.19
-mdformat-mkdocs==3.1.1
+mdformat-mkdocs==4.0.0
 mdformat-admon==2.0.6
```

Comment:
You will need to remove all references to `mdformat-admon` because the mkdocs style for admonitions differs from the python-markdown version

```suggestion
```

Reply:
We have a custom plugin (https://github.com/astral-sh/ruff/blob/4efbb90719a661226b455d682f5c11f6e3fa768e/scripts/_mdformat_utils.py) which was added a while ago which utilizes this as a library. I'll need to look into why that is so and if we can remove it. I don't think that should affect this but I might be wrong.

cc @augustelalande who (I think) wrote that script.

Reply:
You may not need that custom script anymore because the latest version of `mdformat` reduces false positives for text escaping.

You can verify with:

```sh
> echo 'this > that or [that]' | uvx --with='mdformat==0.7.18' mdformat -
this \< that or \[that\]
> echo 'this > that or [that]' | uvx --with='mdformat==0.7.19' mdformat -
this > that or [that]
```

Reply:
I tested changes locally and pre-commit and the generate_mkdocs scripts both work without issue after these changes: https://github.com/astral-sh/ruff/pull/15020/files